src/caret_analyze/infra/lttng/lttng.py

```diff
@@ -329,12 +371,60 @@ def _parse_lttng_data(
 
             handler = Ros2Handler(data, offset)
 
+            # トレースイベントの処理順が以下の様になっている必要がある
+            # (1)初期化系トレースイベントを最初にまとめて処理し、その後ランタイム系トレースイベントを処理する
+            # (2)初期化系トレースイベントを処理する際、あらかじめタイムスタンプでソートする。
+            #    タイムスタンプが同じ場合は、トレースイベントの種類により並び順を決める。
+            #    その際、依存関係の上位にある種類のイベントがより手前に来るようにする。
+            #    これは、ほぼ同時に発生した初期化系トレースポイントが同じタイムスタンプとなった場合への配慮である。
+            # init_events: list[dict] = {}
+            # run_events: list[dict] = {}
+            init_events = []
+            run_events = []
+
+            # 全てのトレースイベントを init_events, run_eventsに振り分ける
+            init_event_names = set(Lttng._prioritized_init_events)
+            for event in event_collection:
+                event_name = event[LttngEventFilter.NAME]
+                if event_name in init_event_names:
+                    init_events.append(event)
+                else:   # MYK test
+                    run_events.append(event)
+
+            # init_events をソートする
+            import functools
+            init_events.sort(key=functools.cmp_to_key(Lttng._compare_init_event))
+            # 初期化系トレースイベントを処理する
+            handler.create_init_handler_map()
+            for event in tqdm(
+                    iter(init_events),
+                    total=len(init_events),
+                    desc='loading',
+                    mininterval=1.0):
+                event_name = event[LttngEventFilter.NAME]
+                if event_name in handler.handler_map:
+                    handler_ = handler.handler_map[event_name]
+                    handler_(event)
```

Comment:
I suggest to remove `if` statement because it should be always true.

```suggestion
                handler_ = handler.handler_map[event_name]
                handler_(event)
```

Reply:
I removed the if statement